pyrival/graphs/maximum_matching.py

```diff
@@ -0,0 +1,65 @@
+import itertools
+from random import randint
+
+
+_DEFAULT_PRIME = 1073750017
+
+
+def maximum_matching(edges, mod=_DEFAULT_PRIME):
+    """
+    Returns the maximum cardinality matching of an undirected general graph
+    Uses a randomized algorithm to compute the rank of the Tutte matrix
+    The rank of the Tutte matrix is equal to twice the size of the maximum matching with high probability
```

Comment:
Would be interesting to know more exactly what the probability is. How does it depend on n?

Reply:
This is nicely explained here - https://www.imsc.res.in/~meena/matching/lecture11.pdf

The probability of error can be proved to be not more than `n*(m+1)/p` although in practice it is much lower.

Reply:
I think you are mixing it up. For one thing the probability n*(m+1)/p in those lecture notes is never stated to be the failure probability of the rank maximum matching algorithm. Secondly, for example https://www.mimuw.edu.pl/~mucha/pub/mucha_sankowski_focs04.pdf states in Theorem 2.3 that the failure probability is n/p. So I think the probability of error is actually <= n / p.

Reply:
You're right, thanks for the reference. I learnt this algorithm a long time ago and didn't keep proper notes. Updated the doc.